src/commands/mod.rs

```diff
@@ -0,0 +1,39 @@
+use clap::{ArgEnum, Parser};
+use std::fmt::{Display, Formatter};
+
+#[derive(Parser)]
+pub struct Input {
+    #[clap(arg_enum)]
+    pub cmd: CargoCommand,
+    pub cargo_opts: Vec<String>,
```

Comment:
How does this behave with an invocation like the following?
```shell
cargo 3ds run --example hello-world -- --some-arg foo
```

I'd love to be able to use `--` to separate executable args from cargo (or cargo-3ds) args, so we can pass them through via 3dslink. Can/does clap support this case? If I recall from my original investigation, it wasn't totally supported but maybe we could do kind of a custom parse to "split" on the `--` ourselves?

Reply:
Since the list is being directly passed to cargo, it actually requires the `--`. At least for some of the flags I tested it cargo responded saying they were necessary. Will have to test more I think to confirm for all but running `--no-run` required me to add the separator.

Reply:
Playing around with it a bit in the current state, running `./cargo-3ds build --example hello-world` would print this error from cargo.

```
error: Found argument '--example' which wasn't expected, or isn't valid in this context

        If you tried to supply `--example` as a value rather than a flag, use `-- --example`

USAGE:
    cargo-3ds.exe <CMD>
```

Reply:
Hmm, I think this change breaks current workflows then – `cargo 3ds [build | run] --example hello-world` works today, so I think it would be preferable to keep that functioning like it has been. (related but outside the scope of this PR: we should probably have some tests to verify workflows actually work).

I found this in another project, which looks like it would be similar to our use case: https://github.com/sharplet/rust-dotenv/commit/8d2ffc011a7cf853c0ff10d21a89eefec17c217f

Maybe there is a `derive` way to do the same there?

Reply:
I think this is resolved